x/subscription/keeper/subscription_test.go

```diff
@@ -581,14 +528,66 @@ func TestExpiryTime(t *testing.T) {
 
 			sub, found := keeper.GetSubscription(ts.ctx, creator)
 			require.True(t, found)
-			require.Equal(t, res, time.Unix(int64(sub.MonthExpiryTime), 0).UTC())
 			require.Equal(t, tt.months, sub.DurationTotal)
 
-			keeper.RemoveSubscription(ts.ctx, creator)
+			// will expire and remove
+			timestamp := ts.ctx.BlockTime()
+			ts.advanceMonths(timestamp, int(tt.months))
+			ts.advanceUntilStale()
 		})
 	}
 }
 
+func TestSubscriptionExpire(t *testing.T) {
+	ts := setupTestStruct(t, 1)
+	keeper := ts.keepers.Subscription
+
+	_, account := sigs.GenerateFloatingKey()
+	coins := sdk.NewCoins(sdk.NewCoin(epochstoragetypes.TokenDenom, sdk.NewInt(10000)))
+	ts.keepers.BankKeeper.SetBalance(ts.ctx, account, coins)
+
+	creator := account.String()
+	consumer := account.String()
+
+	blocksToSave, err := ts.keepers.Epochstorage.BlocksToSave(ts.ctx, uint64(ts.ctx.BlockHeight()))
+	require.Nil(t, err)
+
+	err = keeper.CreateSubscription(ts.ctx, creator, consumer, "mockPlan1", 1)
+	require.Nil(t, err)
+
+	block := uint64(ts.ctx.BlockHeight())
+	timestamp := ts.ctx.BlockTime()
+
+	// fill memory
+	for uint64(ts.ctx.BlockHeight()) < blocksToSave {
+		ts.advanceBlock()
+	}
+
+	_, found := keeper.GetSubscription(ts.ctx, account.String())
+	require.True(t, found)
+
+	err = keeper.ChargeComputeUnitsToSubscription(ts.ctx, account.String(), block, 10)
+	require.Nil(t, err)
+
+	// fast-forward one month
+	ts.advanceMonths(timestamp, 1)
+	_, found = keeper.GetSubscription(ts.ctx, creator)
+	require.False(t, found)
+
+	// subscription no longer searchable, but can still charge for previous usage
+	_, found = keeper.GetSubscription(ts.ctx, account.String())
```

Comment:
you perform this exact check 3 lines above

Reply:
fixed: 45467498